src/timesync/timesyncd-bus.c

```diff
@@ -23,24 +42,86 @@ static int property_get_servers(
                 void *userdata,
                 sd_bus_error *error) {
 
-        ServerName *p, **s = userdata;
-        int r;
+        ServerName **s = userdata;
 
         assert(s);
         assert(bus);
         assert(reply);
 
-        r = sd_bus_message_open_container(reply, 'a', "s");
+        return reply_server_names(*s, reply);
+}
+
+static int property_get_runtime_servers(
+                sd_bus *bus,
+                const char *path,
+                const char *interface,
+                const char *property,
+                sd_bus_message *reply,
+                void *userdata,
+                sd_bus_error *error) {
+
+        Manager *m = userdata;
+
+        assert(m);
+
+        return reply_server_names(m->runtime_servers, reply);
+}
+
+static int property_set_runtime_servers(
+                sd_bus *bus,
+                const char *path,
+                const char *interface,
+                const char *property,
+                sd_bus_message *message,
+                void *userdata,
+                sd_bus_error *error) {
+
+        _cleanup_strv_free_ char **names = NULL;
+        Manager *m = userdata;
+        char **name;
+        int r;
+
+        assert(m);
+        assert(bus);
+        assert(message);
+
+        r = sd_bus_message_read_strv(message, &names);
         if (r < 0)
                 return r;
 
-        LIST_FOREACH(names, p, *s) {
-                r = sd_bus_message_append(reply, "s", p->string);
+        while (m->runtime_servers)
+                server_name_free(m->runtime_servers);
+
+        STRV_FOREACH(name, names) {
+                r = dns_name_is_valid_or_address(*name);
                 if (r < 0)
-                        return r;
+                        return log_error_errno(r, "Failed to check validity of NTP server name or address '%s': %m", *name);
+
+                if (r == 0) {
+                        log_error("Invalid NTP server name or address, ignoring: %s", *name);
+                        continue;
+                }
+
+                r = server_name_new(m, NULL, SERVER_RUNTIME, *name);
+                if (r < 0)
+                        return log_error_errno(r, "Failed to add runtime server '%s': %m", *name);
         }
 
-        return sd_bus_message_close_container(reply);
+        if (m->runtime_servers)
+                m->runtime_servers_changed = true;
+
+        if (manager_is_connected(m)) {
+                if (m->current_server_address &&
+                    m->current_server_address->name &&
+                    m->current_server_address->name->type == SERVER_RUNTIME &&
+                    !strv_find(names, m->current_server_address->name->string)) {
+                        manager_connect(m);
+                }
+        } else {
+                manager_connect(m);
```

Comment:
if you ignore the return value knowingly (which might be OK) then cast the return value to (void) explicitly, to make this clear (here and above)